homeassistant/components/rfxtrx/config_flow.py

```diff
@@ -1,12 +1,281 @@
+from .light import supported as light_supported
+from .switch import supported as switch_supported
 
 _LOGGER = logging.getLogger(__name__)
 
+CONF_OFF_DELAY_ENABLED = CONF_OFF_DELAY + "_enabled"
+CONF_REMOVE_DEVICE = "remove_device"
+
+
+class OptionsFlow(config_entries.OptionsFlow):
+    """Handle Rfxtrx options."""
+
+    def __init__(self, config_entry: ConfigEntry) -> None:
+        """Initialize rfxtrx options flow."""
+        self._config_entry = config_entry
+        self._global_options = None
+        self._selected_device = None
+        self._selected_device_event_code = None
+        self._rfxobj = None
+        self._device_entries = None
+
+    async def async_step_init(self, user_input=None):
+        """Manage the options."""
+        return await self.async_step_prompt_options()
+
+    async def async_step_prompt_options(self, user_input=None):
+        """Prompt for options."""
+        errors = {}
+
+        if user_input is not None:
+            self._global_options = {
+                CONF_DEBUG: user_input[CONF_DEBUG],
+                CONF_AUTOMATIC_ADD: user_input[CONF_AUTOMATIC_ADD],
+            }
+            if CONF_DEVICE in user_input:
+                device_data = self._get_device_data(user_input[CONF_DEVICE])
+                if not device_data.get("event_code"):
+                    errors = {"base": "unknown_event_code"}
+                else:
+                    event_code = device_data["event_code"]
+                    self._selected_device_event_code = event_code
+                    self._selected_device = self._config_entry.data[CONF_DEVICES][
+                        event_code
+                    ]
+                    return await self.async_step_set_device_options()
+            if CONF_REMOVE_DEVICE in user_input:
+                device_data = self._get_device_data(user_input[CONF_REMOVE_DEVICE])
+                if not device_data.get("event_code"):
+                    errors = {"base": "unknown_event_code"}
+                else:
+                    event_code = device_data["event_code"]
+                    device_id = device_data["device_id"]
+                    self.hass.helpers.dispatcher.async_dispatcher_send(
+                        f"{DOMAIN}_{device_id}"
+                    )
+                    device_registry = await async_get_registry(self.hass)
+                    device_registry.async_remove_device(user_input[CONF_REMOVE_DEVICE])
+                    devices = {event_code: None}
+                    self.update_config_data(
+                        global_options=self._global_options, devices=devices
+                    )
+
+                    return self.async_create_entry(title="", data={})
+            if CONF_DEVICE_ID in user_input:
+                self._selected_device_event_code = user_input[CONF_DEVICE_ID]
+                self._selected_device = {}
+                return await self.async_step_set_device_options()
```

Comment:
We should validate that the event_code is parsable here. So i think we should set the _rfxobj before we go to device_options.